### PR TITLE
Prepare for 4.0.0

### PR DESCRIPTION
**Pull request by nirinchev**
🎉

### CODE REVIEW

**Comment by nirinchev** [comment]
@bmunkholm feel free to review the changelog here: https://github.com/realm/realm-dotnet/blob/fe10a22a284302d439b9ebdcc4f3e48aa7ff224a/CHANGELOG.md

**Comment by bmunkholm** [comment]
Isn't ROS 3.21 required? 
Could we add an internal section in the release note listing the sync version as we do for other SDKs now?
Haven't we fixed bugs through upgraded sync version?

**Comment by nirinchev** [comment]
@bmunkholm I've updated the ROS requirement - it's actually 3.23.1 and added changelog entries for the fixes from Core/Sync - I didn't find any sync fixes that affect the SDKs directly, but I added the ones from Core that seemed relevant.

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,12 +61,21 @@
...
 * Added `RealmOwnerId` property to `PathPermission` that indicates who the owner of the Realm is. ([Issue #1863](https://github.com/realm/realm-dotnet/issues/1863))
+* Added support for building with `dotnet build` (previously only the `msbuild` command line was supported). ([PR #1849](https://github.com/realm/realm-dotnet/pull/1849))
+* Improved query performance for unindexed string columns when the query has a long chain of OR conditions. (Core upgrade)
+* Improved performance of encryption and decryption significantly by utilizing hardware optimized encryption functions. (Core upgrade)
+* Compacting a realm into an encrypted file could take a really long time. The process is now optimized by adjusting the write buffer size relative to the used space in the realm. (Core upgrade)
+* The string-based query parser (`results.Filter("...")`) now supports readable timestamps with a 'T' separator in addition to the originally supported "@" separator. For example: `"startDate > 1981-11-01T23:59:59:1"`. (Core upgrade)
```
Comment (bmunkholm) [inline CHANGELOG.md]: Forgive my ignorance - does .Net also support string based queries now?

**Comment by bmunkholm** [comment]
Do we have a PR or Issue for doc updates?

File: CHANGELOG.md (hunk 1 above)
Comment (nirinchev) [reply CHANGELOG.md]: Yes - we added support in 3.0.0 as we didn't want to hold functionality hostage to our LINQ support: https://github.com/realm/realm-dotnet/blob/master/CHANGELOG.md#enhancements-5

**Comment by nirinchev** [comment]
Here's the PR for docs: https://github.com/realm/realm.io/pull/4641

File: CHANGELOG.md (hunk 1 above)
Comment (bmunkholm) [reply CHANGELOG.md]: Cool - Where does one find docs for that? (I did actually try to find it without success)

**Comment by nirinchev** [comment]
@bmunkholm the .net API docs have GitHub links for method implementations that unfortunately can't resolve tags (they only work with branches) so we keep the release branches around.

**Pull request merged**